module/zfs/vdev_queue.c

```diff
@@ -803,6 +803,48 @@ vdev_queue_io_done(zio_t *zio)
 	mutex_exit(&vq->vq_lock);
 }
 
+void
+vdev_queue_change_io_priority(zio_t *zio, zio_priority_t priority)
+{
+	vdev_queue_t *vq = &zio->io_vd->vdev_queue;
+	avl_tree_t *tree;
+
+	ASSERT3U(zio->io_priority, <, ZIO_PRIORITY_NUM_QUEUEABLE);
+	ASSERT3U(priority, <, ZIO_PRIORITY_NUM_QUEUEABLE);
+
+	if (zio->io_type == ZIO_TYPE_READ) {
```

Comment:
Can you explain why we need this? And also when would we want to change the priority of a WRITE?

Reply:
The block of code is there because it is also in `vdev_queue_io()` and I wanted the same semantics. I don't know when we would currently want to change the priority of a write, but I didn't see any reason to exclude it either.

Reply:
I'm not sure we want to add code that is never called.

Reply:
Would you prefer me to ASSERT that the call is always a read?

Reply:
Yes, I think you can rip this whole block out and add an ASSERT.